src/AutoRest.CSharp/Mgmt/Output/Resource.cs

```diff
@@ -451,31 +458,64 @@ protected virtual FormattableString CreateDescription()
             _ => FormattableStringHelpers.Join(parentTypes.Select(type => (FormattableString)$"<see cref=\"{type}\" />").ToList(), ", ", " or "),
         };
 
-        private static Type? ToFormatTypeByName(string? name) => name switch
-        {
-            "location" => typeof(AzureLocation),
-            _ => null
-        };
+        private static CSharpType GetReferenceType(Reference reference)
+            => reference.Name switch
+            {
+                "location" when reference.Type.EqualsIgnoreNullable(typeof(string)) => typeof(AzureLocation),
+                _ => reference.Type
```

Comment:
Do we really expect any type here, or those are limited to strings?

Reply:
Well the old code is matching parameter called "location" with any type.
But since these are path parameters, there is really a very limited set of types it could choose from  - integers and strings basically.
It works perfectly fine if I remove the constraint but I just think that AzureLocation, this type by design only takes in a `string` as input, therefore I added the constraint.